pyccel/ast/core.py

```diff
@@ -3371,8 +3376,56 @@ def attributes_as_dict(self):
             d_attributes[i.name] = i
         return d_attributes
 
-    # TODO add other attributes?
+    def add_new_attribute(self, attr):
+        """
+        Add new attributes to the current class.
+
+        A new attributes of the current ClassDef.
+
+        Parameters
+        ----------
+        attr : Varibale
+            The Variable that will be added.
+        """
+
+        if not isinstance(attr, Variable):
+            raise TypeError("Attributes must be Variables")
+        attr.set_current_user_node(self)
+        self._attributes += (attr,)
+
+    def add_new_method(self, method):
+        """
+        Add a new method to the current class.
+
+        A new method of the current ClassDef.
+
+        Parameters
+        ----------
+        method : FunctionDef
+            The Method that will be added.
+        """
+
+        if not isinstance(method, FunctionDef):
+            raise TypeError("Method must be FunctionDef")
+        method.set_current_user_node(self)
+        self._methods += (method,)
+
+    def add_new_interface(self, interface):
+        """
+        Add a new interface to the current class.
+
+        A new interface of the current ClassDef.
+
+        Parameters
+        ----------
+        interface : FunctionDef
+            The interface that will be added.
+        """
 
+        if not isinstance(interface, Interface):
+            raise TypeError("Interface must be Interface")
```

Comment:
The error message is a bit cryptic. I suggest
```suggestion
            raise TypeError("Argument "interface" must be of type Interface")
```